src/lib/browse/LayerControls.svelte

```diff
@@ -72,6 +73,28 @@
       return getRoadLayerNames(map, get(mapStyle));
     },
   };
+
+  const initialQueryParams = new URLSearchParams(window.location.search);
+
+  if (
+    !initialQueryParams ||
+    (initialQueryParams.size === 1 &&
+      initialQueryParams.keys().next().value == "style")
+  ) {
+    let url = new URL(window.location.href);
+    const previousSessionQueryString =
+      window.localStorage.getItem(localStorageQuerystringKey) || "";
+    const newQueryParams = new URLSearchParams(previousSessionQueryString);
+    // @ts-ignore typescript is convinced foreach doesn't exist but it's working?
```

Comment:
Always prefer `ts-expect-error`, because if the code does _not_ have any TS error, then the annotation causes an error. `ts-ignore` could wind up being a stray annotation silently not doing anything.